### PR TITLE
[stable/influxdb] changes from deployment to statefulset

### PR DESCRIPTION
**Pull request by felippe-mendonca**
#### What this PR does / why we need it:
Since influxdb can use storage to persist its data, using a StatefulSet is better than a Deployment with a volume. Some advantages:

- There is no need to handle with using created PV. Using a persistent storage-class, even if you delete the StatefulSet, it'll be re-attached to the Pod when you re-reploy.
- PVC manifest is not necessary.
- StatefulSets guarantee that when deleting the pod, the PV associated with it is not deleted by default.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/influxdb/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: influxdb
-version: 1.4.0
+version: 1.5.0
```
Comment (Aisuko) [inline stable/influxdb/Chart.yaml]: Thanks for your PR, so cleanly. But there may have some considerations on the chart version. This PR include don't backward compatibility. We should bump the chart version to `2.0.0` and make tips on README.md.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for your PR, there may some suggestions.

File: stable/influxdb/Chart.yaml (hunk 1 above)
Comment (felippe-mendonca) [reply stable/influxdb/Chart.yaml]: Totally agree with you. I'm going to work on that soon!

**Comment by zanhsieh** [comment]
@felippe-mendonca 
Would you resolve the conflicting files please?
stable/influxdb/Chart.yaml
stable/influxdb/templates/deployment.yaml

**Comment by felippe-mendonca** [comment]
Hi @aisuko, sorry the late! I rebased and made the requested changes. Although, I notice that Chart version had already been bumped. So I just increase a minor version, from `2.0.1` to `2.1.0`. Is that ok?

File: stable/influxdb/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: influxdb
-version: 2.0.1
+version: 2.1.0
```
Comment (Aisuko) [inline stable/influxdb/Chart.yaml]: I believe that bump the chart version to `3.0.0` can make more sense.

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for your PR, I believe we can merge this after bump the chart version.

**Review by Aisuko — APPROVED** [review]
/LGTM

**Comment by Aisuko** [comment]
Thanks a lot.

**Pull request merged**